PR: Improve dialog to create new projects

ccordoba12 opened the pull request:
## Description of Changes

- The dialog was reimplemented as a `SidebarDialog` to make it easier to follow and have an enhanced UI.
- This will help to extend the dialog with new improvements for remote development.
- Rename `ValidationLabel` (in the Remote client plugin) to `MessageLabel` and add it to `helperwidgets` to use it in this dialog.
- Use home directory only when selecting files and directories in `SpyderConfigPage`. That avoids using the cwd where Spyder was started, which could not be the best directory to use in this case.
- Add class constant to `SidebarDialog` to be of fixed size.

### Visual changes

| Before | After |
| - | - |
| ![imagen](https://github.com/user-attachments/assets/6a557a3a-7daf-4117-9a77-855d8fdf67ce) | ![imagen](https://github.com/user-attachments/assets/ee68f79c-45ef-4569-ad5c-b24d205cfe73) |

### Issue(s) Resolved

<!--- List the issue(s) below, in the form "Fixes #1234"; one per line --->

Fixes #


### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

<!--- TYPE YOUR USER/NAME AFTER THE FOLLOWING: --->
I certify the above statement is true and correct: @ccordoba12 

<!--- Thanks for your help making Spyder better for everyone! --->

ccordoba12 commented:
@dalthviz, this is ready for review. Please check that project validations work as before and that you can create projects in new and existing directories with the new UI.

dalthviz reviewed (commented):
Thanks @ccordoba12 ! I gave this a check locally on Windows and for the most part seems like is working correctly :+1:

Besides a couple of suggestions to ensure the link to the projects docs page works and uses `https`, I noticed that there is a error message that I'm not totally sure to follow. When trying to click `Create` without any input done yet a couple of messages appear:

![new_project_dialog](https://github.com/user-attachments/assets/1b0cbecb-53e0-40f2-9ba0-838bddac858d)

Is the message related with the selected directory for this project already existing correct? Shouldn't it be something like **_You need to selected a parent directory for the project directory_** or something like that?

Also, seems like the widget to check for the location can break with certain paths? For example, if I put `/users`:

![new_project_dialog_2](https://github.com/user-attachments/assets/97ed10c2-3b99-40c3-bf75-e10210204394)

Besides that, and thinking on the info available from the previous dialog, could it make sense to add a disabled label that shows the full path of the new project bellow the Project name and Location widgets? Something like `Project path`?

ccordoba12 commented:
@dalthviz, this is ready for another review.

> I noticed that there is an error message that I'm not totally sure to follow. When trying to click Create without any input done yet a couple of messages appear

> Is the message related with the selected directory for this project already existing correct?

Sorry, that was due to some changes related to `os.path.normpath` that I had to do to make the tests pass. But it should be fixed now.

> Also, seems like the widget to check for the location can break with certain paths? For example, if I put /users

Thanks for checking that! I changed the implementation to check if a directory is writable to fix that and added a test for it too.

dalthviz commented:
Checked again and seems like the error messages shown make sense :+1: Last couple of things that I noticed:

* The dialog doesn't have an icon:

![image](https://github.com/user-attachments/assets/62ceb87a-ab2a-42a5-8b92-3f257df7a9fb)

* The fixed dialog size causes the options labels to be shown elided (more noticeable when using for example Spanish as language and maybe due to my screen resolution?):

![image](https://github.com/user-attachments/assets/f5fd10aa-9605-4404-84d3-097556142962)

Other than that this LGTM :+1:

ccordoba12 commented:
> The dialog doesn't have an icon

Good catch! I added an icon for it in my last commit.

> The fixed dialog size causes the options labels to be shown elided (more noticeable when using for example Spanish as language and maybe due to my screen resolution?)

Actually, it's not the fixed size for the dialog that causes that but the width of the widget that shows the entries on the left. It needs to be fixed to correctly display the section separators in Preferences:

![imagen](https://github.com/user-attachments/assets/1771fbee-b42c-4980-b497-05889b0782fa)

https://github.com/spyder-ide/spyder/blob/b117c4456200cc8aa25aca46e3a0c7663386673c/spyder/widgets/sidebardialog.py#L459-L472

For now I'm going to leave things like that, but I'll try to revisit them in a follow-up PR.

ccordoba12 commented:
Merging as per last @dalthviz's comment: https://github.com/spyder-ide/spyder/pull/24249#issuecomment-2840349870
